blockchain/fastsync.go

```diff
@@ -213,38 +213,69 @@ func worker(cache *utxoCache, jobs <-chan []byte, results chan<- *result) {
 // downloadUtxoSet will attempt to connect to make an HTTP GET request to the
 // provided sources one at a time and download the UTXO set to the provided path.
 // If a proxy is provided it will use it for the HTTP connection.
-func downloadUtxoSet(sources []string, proxy *socks.Proxy, pth string) (string, error) {
-	var fileName string
-
-	grab.SetDefaultClientWithProxy(proxy)
-
-	for _, src := range sources {
-		log.Infof("Downloading UTXO set from %s", src)
-		err := retry.Do(
-			func() error {
-				resp, err := grab.Get(pth, src)
-				if err != nil {
-					return err
+func downloadUtxoSet(sources []string, proxy *socks.Proxy, destination string) (string, error) {
+	for _, source := range sources {
+		log.Infof("FastSync: Downloading UTXO set from %s to %s ...", source, destination)
+		retries := 3
+		for retries > 0 {
+			retries--
+
+			client := grab.NewClientWithProxy(proxy)
+			request, _ := grab.NewRequest(destination, source)
+
+			// Begin download
+			response := client.Do(request)
+
+			if response.HTTPResponse != nil {
+				fmt.Printf("FastSync: %v\n", response.HTTPResponse.Status)
+			}
+
+			displayDownloadProgress(response)
+
+			// Error checking
+			if responseError := response.Err(); responseError != nil {
+				log.Errorf("FastSync: Failed downloading UTXO set: %s", responseError.Error())
+				if strings.Contains(responseError.Error(), "connection refused") {
+					break
+				} else {
+					continue
 				}
+			}
 
-				fileName = resp.Filename
+			filename := response.Filename
+			log.Infof("\nFastSync: Download saved to %v", filename)
```

Comment:
I don't think we need a prefixed newline do we? We should always be using `log.Info` or `log.Infof` with trailing newline to keep the logging flow consistent.

Reply:
Sure.
Removed

Reply:
`Infof` should have a trailing `\n` still =)

Reply:
Does it ?I'll take your word for it and add in the \n

Reply:
No need, commented above. =)